src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/multiarch.py

```diff
@@ -0,0 +1,156 @@
+        return 'Build definition for multi-architecture manifest successfully deleted.'
+    raise CLIError("There was an error deleting the resource. " + str(response.status_code))
+
+def _validate_user_credentials(registry_name,
+                               repository,
+                               multiArch_tag,                            
+                               tags,
+                               build_name,
+                               resource_group_name=None, 
+                               username=None,
+                               password=None,
+                               yaml=None):
+
+    registry, _ = get_registry_by_name(registry_name, resource_group_name)
+    location=registry.location
+    login_server = registry.login_server
+    #1. if username was specified, verify that password was also specified
+    if username:
+        if not password:
+            try:
+                password=prompt_pass(msg='Password: ')
+            except NoTTYException:
+                raise CLIError('Please specify both username and password in non-interactive mode.')
+        return create_build(location,registry_name,repository,multiArch_tag,tags,build_name,resource_group_name,username,password,yaml)
+    # 2. if we still don't have credentials, attempt to get the admin credentials (if enabled)
+    try:
+        cred = acr_credential_show(registry_name)
+        username = cred.username
+        password = cred.passwords[0].value
+        return create_build(location,registry_name,repository,multiArch_tag,tags,build_name,resource_group_name,username,password,yaml)
+    except NotFound as e:
+        raise CLIError(str(e))
+    except Unauthorized as e:
+        logger.warning("Unable to authenticate using admin login credentials: %s", str(e))
+    except Exception as e:  # pylint: disable=broad-except
+        logger.warning("Admin user authentication failed with message: %s", str(e))
+    # 3. if we still don't have credentials, prompt the user
+    try:
+        username = prompt('Username: ')
+        password = prompt_pass(msg='Password: ')
+    except NoTTYException:
+        raise CLIError(
+            'Unable to authenticate using AAD tokens or admin login credentials. ' +
+            'Please specify both username and password in non-interactive mode.')
+    return create_build(location,registry_name,repository,multiArch_tag,tags,build_name,resource_group_name,username,password,yaml)
+
+def create_build(location,
+                 registry_name,
+                 repository,
+                 multiArch_tag,                            
+                 tags,
+                 build_name,                 
+                 resource_group_name=None, 
+                 username=None,
+                 password=None,
+                 yaml=None):
+
+    resource_group_name=get_resource_group_name_by_registry_name(registry_name);
+    if not yaml:
+        tagPrefix=registry_name+'.azurecr.io/'+repository+':';
```

Comment:
Get login server from registry. `.azurecr.io` may not always be the domain suffix.